homeassistant/components/goalzero/__init__.py

```diff
@@ -78,14 +80,23 @@ def __init__(self, api, coordinator, name, server_unique_id):
         self._name = name
         self._server_unique_id = server_unique_id
         self._device_class = None
+        self._model = None
 
     @property
     def device_info(self):
         """Return the device information of the entity."""
+        if self.api.data:
+            sw_version = self.api.data["firmwareVersion"]
+        else:
+            sw_version = None
+        if self.api.sysdata:
+            self._model = self.api.sysdata["model"]
```

Comment:
Why do we need to store the model in an instance attribute? Maybe it's enough to use a local variable?

Reply:
The model is not exposed at the `/state` endpoint. It's called once during setup at `/sysinfo`.